src/update.rs

```diff
@@ -50,7 +50,19 @@ pub fn update<T: CachedBackend + ?Sized>(
                 return Err(CliError::InvalidComponentName(comp.clone()))
             }
             // fetch without a specific version (latest)
-            match backend.unpack_published_component(comp, None, Some(env)) {
+
+            // First, since this potentially goes in the manifest
+            // make sure the version is found for all supported environments:
+            let res = backend.get_latest_supported_versions(comp, manifest.supportedEnvironments.clone())?;
+            let ver = match res.into_iter().max() {
+                Some(m) => m,
+                None => {
+                    return Err(CliError::NoIntersectedVersion(comp.clone()));
+                }
+            };
```

Comment:
Sometimes I just enjoy reading rust error handlers.
It's something about knowing that after this line, `ver` is something sensible.

Reply:
If you liked that, replace that entire match with a chained on:

```sh
.ok_or(CliError::NoIntersectedVersion(comp.clone()))?
```

Reply:
... tbh, I prefer it the other way.

Reply:
hah, aw